### PR TITLE
README, config file and maintainer guide updates

### PR DESCRIPTION
**Pull request by filmaj**
###  Summary

- Updating test and coverage tasks to work properly.
- Putting `lint` and `fmt` options into the `deno.jsonc` configuration file.
- Updating the releasing steps to converge around creating GitHub Releases.

Whatever changes we end up converging on here I can then copy over to the sdk, builder, runtime and hooks repos to mirror this. I will then also link to these in my slack-cli PR to update the rather-complex releasing steps therein to link out where appropriate.

### CODE REVIEW

File: .github/maintainers_guide.md [hunk 1]
```diff
@@ -24,23 +24,33 @@ Run this from the root of the project
...
 ### Releasing
 
-* Releases for this library are triggered off of tags. You should create a new GitHub Release, and use a new version that adheres to semantic versioning based on what's being release. Version tags should match the following pattern: `1.0.1` (no `v` preceding the number).
-* Once the release has been created, a corresponding version will be avaliable on https://deno.land/x/deno_slack_api.
+Releases for this library are automatically generated off of git tags. To create a new release:
+
+1. Create a new GitHub Release from the [Releases page](https://github.com/slackapi/deno-slack-api/releases) by clicking the "Draft a new release" button.
+2. Input a new version manually into the "Choose a tag" input. Ensure that this version adheres to [semantic versioning][semver] based on what's being released. Version tags should match the following pattern: `1.0.1` (no `v` preceding the number).
+  - After you input the new version, click the "Create a new tag: x.x.x on publish" button.
+3. Set the "Target" input to the "main" branch.
```
Comment (shapirone) [inline .github/maintainers_guide.md:42]: nit: Should we call out anything about ensuring that everything on `main` is in a releasable state?

File: .github/maintainers_guide.md [hunk 2]
```diff
@@ -24,23 +24,33 @@ Run this from the root of the project
...
+4. Name the release title after the version tag.
+5. You can copy the [release contents of the 0.0.3 release](https://github.com/slackapi/deno-slack-api/releases/edit/0.0.3) to use as a template for the release notes. Make sure to call out the changes under "What's Changed" and shout out any new contributors under "New Contributors". Also don't forget to update the detailed changelog link at the bottom to reference the latest and previous release correctly!
```
Comment (shapirone) [inline .github/maintainers_guide.md:44]: We've actually just been using the "Auto-generate release notes" button that does this for us, I think we can recommend that instead. Maybe call out to review the notes and make sure they're more reader friendly when applicable.

**Review by shapirone — COMMENTED** [review]
🎉 Thanks @filmaj this is really helpful! Added some discussion points, would love to get input from @selfcontained as well

File: .github/maintainers_guide.md (hunk 1 above)
Comment (filmaj) [reply .github/maintainers_guide.md:44]: I added a note about this in the intro sentence to this section.

**Review by shapirone — APPROVED** [review]
LGTM!

File: .github/maintainers_guide.md [hunk 3]
```diff
@@ -4,15 +4,15 @@ This document describes tools, tasks and workflow that one needs to be familiar
...
+## Tools
 
 All you need to work on this project is a recent version of [Deno](https://deno.land/)
 
 ## Tasks
 
 ### Building
 
-* The majority of this library (`src/generated`) is generated code based off of a slack api spec file. Updating the web apis based off a new spec file can be done by getting the latest spec file from https://api.slack.com/web#spec or via the [slack-api-specs](https://github.com/slackapi/slack-api-specs) repo directly. 
+* The majority of this library (`src/generated`) is generated code based off of a slack api spec file. Updating the web apis based off a new spec file can be done by getting the latest spec file from https://api.slack.com/web#spec or via the [slack-api-specs](https://github.com/slackapi/slack-api-specs) repo directly.
```
Comment (selfcontained) [inline .github/maintainers_guide.md:15]: Seeing this white-space change made me realize this is no longer true. This code is generated off of [a file](https://github.com/slackapi/deno-slack-api/blob/main/scripts/src/public-api-methods.ts) that has a list of api methods. I'm happy to update that detail in a followup PR though, as it's been wrong here, and you were just fixing some white-space here 😄

File: deno.jsonc [hunk 4]
```diff
@@ -1,6 +1,16 @@
 {
+  "fmt": {
```
Comment (selfcontained) [inline deno.jsonc:2]: TIL you can set this config up here too, nice!

**Review by selfcontained — APPROVED** [review]
Thanks, this is great! I left a non-blocking comment about us fixing some of the details around the generated code, but can be a followup PR, it's been incorrect for quite awhile.

File: .github/maintainers_guide.md (hunk 3 above)
Comment (filmaj) [reply .github/maintainers_guide.md]: I added a bit of detail in this PR, but if it is incomplete or insufficient, we can follow up with another!

**Pull request merged**